lib/iotagent-mqtt.js

```diff
@@ -144,6 +145,24 @@ function mqttMessageHandler(topic, message) {
 }
 
 /**
+ * Generate the list of topics related to the device, based on the device attribute definitions.
+ *
+ * @param {Object} device           Device object containing all the information about the provisioned device.
+ * @return {Array}                  List of strings names of the topics associated to the selected device.
+ */
+function generateDeviceTopics(device) {
+    var topics = [];
+
+    for (var i = 0; i < device.active.length; i++) {
```

Comment:
Does `device.active` always exist and it is an `Array` (it could be an empty one)? If `device.active` was `undefined`, this line may end up throwing an `Error`... Just in case you want to cross-check it and leave as it is right now or include an `Array,isArray()` check to be 100% sure... :) I leave it to your discretion... ;)